solr/core/src/java/org/apache/solr/core/ConfigSetService.java

```diff
@@ -56,11 +58,24 @@
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
   public static ConfigSetService createConfigSetService(NodeConfig nodeConfig, SolrResourceLoader loader, ZkController zkController) {
-    if (zkController == null) {
+
+    String configSetServiceClass = nodeConfig.getConfigSetServiceClass();
+
+    if(configSetServiceClass != null){
+      try {
+        Class<? extends ConfigSetService> clazz = loader.findClass(configSetServiceClass, ConfigSetService.class);
+        Constructor<? extends ConfigSetService> constructor
```

Comment:
I would rather we pass the NodeConfig instead of the boolean and let the implementation examine whatever it may want.  I think that's easier to maintain.

Reply:
do you mean that make the examining logic whether should use custome ConfigSetService creation  "configSetServiceClass != null"  move into NodeConfig?  And the creation of custome ConfigSetService instance should be move to other class?

Reply:
No.  I'm referring to the constructor signature of ConfigSetService.  Replace the Boolean (for schema caching) with NodeConfig.  If the ConfigSetService cares to honor that option (or others), it's up to it.

Reply:
OK，I got it.  that's easy